From 91b26cccb4d26fec4d3c987a000294ad8123f672 Mon Sep 17 00:00:00 2001
From: Wolfgang Bauer <wbauer@tmo.at>
Date: Mon, 10 Aug 2020 14:02:42 +0000
Subject: [PATCH] Fix build with Qt versions before 5.12

The used QWheelEvent constructor only exists since Qt 5.12, so use an obsolete one
instead when building with lower versions.

This is basically a conditional revert of (parts of) commit 35f3a486.
---
 ThumbnailView/ThumbnailWidget.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ThumbnailView/ThumbnailWidget.cpp b/ThumbnailView/ThumbnailWidget.cpp
index ab8768ab..773868b7 100644
--- a/ThumbnailView/ThumbnailWidget.cpp
+++ b/ThumbnailView/ThumbnailWidget.cpp
@@ -211,8 +211,13 @@ void ThumbnailView::ThumbnailWidget::wheelEvent(QWheelEvent *event)
         cellGeometryInfo()->calculateCellSize();
         model()->endResetModel();
     } else {
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 0)
+        const int delta = event->delta() / 5;
+        QWheelEvent newevent = QWheelEvent(event->pos(), delta, event->buttons(), event->modifiers());
+#else
         const auto angleDelta = event->angleDelta() / 5;
         QWheelEvent newevent = QWheelEvent(event->pos(), event->globalPos(), event->pixelDelta(), angleDelta, event->buttons(), event->modifiers(), event->phase(), event->inverted());
+#endif
 
         QListView::wheelEvent(&newevent);
         event->setAccepted(newevent.isAccepted());
-- 
GitLab

